Server/Controllers/TransactionController.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using AutoMapper;
+using Endava_Project.Server.Data;
+using Endava_Project.Server.Models;
+using Endava_Project.Shared;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace Endava_Project.Server.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class TransactionController : ControllerBase
+    {
+        private readonly ApplicationDbContext context;
+        private readonly UserManager<ApplicationUser> userManager;
+
+        public TransactionController(ApplicationDbContext context, UserManager<ApplicationUser> userManager)
+        {
+            this.context = context;
+            this.userManager = userManager;
+        }
+
+        [HttpGet]
+        [Route("{typeFilter}/{sortFilter}/{orderFilter}/{itemsPerPage}/{pageNumber}")]
+        public TransactionHistoryData GetTransactions(string typeFilter,string sortFilter,string orderFilter, int itemsPerPage, int pageNumber)
+        {
+            var transactionData = new TransactionHistoryData { 
+                TransactionsList = new List<TransactionDto>()
+            };
+            var transactionsList = new List<Transaction>();
+            var userId = userManager.GetUserId(User);
+            var idList = context.Users.Include(e => e.Wallets).FirstOrDefault(x => x.Id == userId).Wallets.Select(e => e.Id).ToList();
```

Comment:
I believe if Wallet had an owner property (User), then we could do something like this:
`context.Wallets.Where(x => x.Owner.Id == userId).Select(x => x.Id).ToList()`
\+ your query will fail if FirstOrDefault returns null